pkgs/applications/misc/iterm2/default.nix

```diff
@@ -1,31 +1,48 @@
 { stdenv, fetchFromGitHub }:
 
+ /*
+ This derivation is impure: it relies on an Xcode toolchain being installed
+ and available in the expected place. The values of sandboxProfile
+ are copied pretty directly from the MacVim derivation, which
+ is also impure. In order to build you at least need the `build-use-chroot`
+ option set to `relaxed` or set the `sandbox` option to `false`.
+ */
+
 stdenv.mkDerivation rec {
   pname = "iterm2";
-  version = "3.0.14";
+  version = "3.3.9";
 
   src = fetchFromGitHub {
     owner = "gnachman";
     repo = "iTerm2";
     rev = "v${version}";
-    sha256 = "03m0ja11w9910z96yi8fzq3436y8xl14q031rdb2w3sapjd54qrj";
+    sha256 = "06mq3gfjgy8jw2f3fzdsi3pbfkdijfzzlhlw6ixa5bfb4hbcgn5j";
   };
 
   patches = [ ./disable_updates.patch ];
   postPatch = ''
     sed -i -e 's/CODE_SIGN_IDENTITY = "Developer ID Application"/CODE_SIGN_IDENTITY = ""/g' ./iTerm2.xcodeproj/project.pbxproj
   '';
+
   preConfigure = "LD=$CC";
-  makeFlagsArray = ["Deployment"];
+  makeFlagsArray = ["Nix"];
   installPhase = ''
-    mkdir -p "$out/Applications"
-    mv "build/Deployment/iTerm2.app" "$out/Applications/iTerm.app"
+    mkdir -p $out/Applications
+    mv Build/Products/Deployment/iTerm2.app $out/Applications/iTerm.app
+  '';
+
+  sandboxProfile = ''
```

Comment:
What is `sandboxProfile` and where is it documented?

Reply:
The `sandboxProfile` allows to make an impure derivation a bit _purer_ by specifying some access (read, write) rules. As mentioned it was copied from the `macvim`derivation and they state that it does the following:
> We rely on the user's Xcode install to build. It may be located in an arbitrary place, and
  it's not clear what system-level components it may require, so for now we'll just allow full
  filesystem access. This way the package still can't access the network.